osquery/tables/utility/file.cpp

```diff
@@ -54,7 +54,7 @@ void genFileInfo(const fs::path& path,
     // Path was not real, had too may links, or could not be accessed.
     return;
   }
-  if ((link_stat.st_mode & S_IFLNK) != 0) {
+  if ((link_stat.st_mode & S_IFLNK) == S_IFLNK) {
```

Comment:
nit: this should work but can we use S_ISLNK() macro instead?

Reply:
It looks like it was using S_ISLNK() but was changed in this PR https://github.com/facebook/osquery/pull/2286